monitor/puller.go

```diff
@@ -101,6 +103,19 @@ func (m *Monitor) syncIntendWithdrawChannel(intendWithdrawChannel *mainchain.Cel
 	}
 }
 
+func (m *Monitor) syncSGNAddSubscriptionBalance(event *mainchain.SGNAddSubscriptionBalance) {
+	consumer := event.Consumer
+	amount := event.Amount
+	log.Infof("Sync AddSubscriptionBalance for %x, amount %s, tx hash %x", consumer, amount, event.Raw.TxHash)
+	transactor := m.Transactor
+	subscription := guard.NewSubscription(consumer.Hex())
+	subscription.Deposit = sdk.NewIntFromBigInt(amount)
+	subscriptionData := transactor.CliCtx.Codec.MustMarshalBinaryBare(subscription)
+	msg := sync.NewMsgSubmitChange(sync.Subscribe, subscriptionData, m.Transactor.Key.GetAddress())
+	log.Infof("submit change tx: subscribe EthAddress %s, Amount %s", consumer, amount)
+	transactor.AddTxMsg(msg)
+}
```

Comment:
need to check if the info has already been updated by the previous puller. Please check other puller funcs (e.g., `SyncValidator`, `SyncUpdateSidechainAddr`) for examples.

Reply:
Done.